examples/pytorch_lightning_simple.py

```diff
@@ -81,58 +68,80 @@ def forward(self, data):
+    def validation_step(self, batch, batch_idx: int) -> None:
         data, target = batch
-        output = self.forward(data)
+        output = self(data)
         pred = output.argmax(dim=1, keepdim=True)
         accuracy = pred.eq(target.view_as(pred)).float().mean()
         self.log("val_acc", accuracy)
 
-    def configure_optimizers(self):
-        return Adam(self.model.parameters())
+    def configure_optimizers(self) -> optim.Optimizer:
+        return optim.Adam(self.model.parameters())
+
 
-    def train_dataloader(self):
-        return torch.utils.data.DataLoader(
-            datasets.MNIST(DIR, train=True, download=True, transform=transforms.ToTensor()),
-            batch_size=BATCHSIZE,
-            shuffle=True,
+class MNISTDataModule(pl.LightningDataModule):
+    def __init__(self, data_dir: str, batch_size: int):
+        super().__init__()
+        self.data_dir = data_dir
+        self.batch_size = batch_size
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        self.mnist_test = datasets.MNIST(
+            self.data_dir, train=False, download=True, transform=transforms.ToTensor()
+        )
+        mnist_full = datasets.MNIST(
+            self.data_dir, train=True, download=True, transform=transforms.ToTensor()
+        )
+        self.mnist_train, self.mnist_val = random_split(mnist_full, [55000, 5000])
+
+    def train_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_train, batch_size=self.batch_size, shuffle=True, pin_memory=True
         )
 
-    def val_dataloader(self):
-        return torch.utils.data.DataLoader(
-            datasets.MNIST(DIR, train=False, download=True, transform=transforms.ToTensor()),
-            batch_size=BATCHSIZE,
-            shuffle=False,
+    def val_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_val, batch_size=self.batch_size, shuffle=False, pin_memory=True
         )
 
+    def test_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_test, batch_size=self.batch_size, shuffle=False, pin_memory=True
+        )
 
-def objective(trial):
-    # Filenames for each trial must be made unique in order to access each checkpoint.
-    checkpoint_callback = pl.callbacks.ModelCheckpoint(
-        os.path.join(MODEL_DIR, "trial_{}".format(trial.number), "{epoch}"), monitor="val_acc"
-    )
+
+def objective(trial) -> float:
+
+    model = LightningNet(trial)
+    datamodule = MNISTDataModule(data_dir=DIR, batch_size=BATCHSIZE)
 
     trainer = pl.Trainer(
         logger=False,
         limit_val_batches=PERCENT_VALID_EXAMPLES,
-        checkpoint_callback=checkpoint_callback,
+        checkpoint_callback=False,
         max_epochs=EPOCHS,
-        gpus=1 if torch.cuda.is_available() else None,
-        callbacks=[PyTorchLightningPruningCallback(trial, monitor="val_acc", mode="max")],
+        gpus=-1 if torch.cuda.is_available() else None,
```

Comment:
I think it's better to let this program use a GPU if available. We can disable the use of GPUs outside the program by using `CUDA_VISIBLE_DEVICES`.
```suggestion
        gpus=1 if torch.cuda.is_available() else None,
```

Reply:
Setting `-1` in PL says that you use all available GPUs :)
https://pytorch-lightning.readthedocs.io/en/stable/trainer.html#gpus
<img width="420" alt="Screenshot 2021-02-08 at 16 09 21" src="https://user-images.githubusercontent.com/7549587/107224167-219b6400-6a28-11eb-91e5-6eab5c8a5d0a.png">

Reply:
Oh, thank you. Obviously I missed that.